src/lib/eliom_comet.client.mli

```diff
@@ -115,17 +110,41 @@ module Configuration : sig
 end
 
 module Channel : sig
-  type 'a t = 'a Lwt_stream.t
+  type 'a t
+
+  type callback_id
+  (** Handler returned by {!register} that allows unregistering a callback
+      later with {!unregister}. *)
+
+  val register : 'a t -> ('a option -> unit Lwt.t) -> callback_id
+  (** [register chan callback] registers a callback to be called for new messages
+    from the server. The callback receives [Some data] for each new messages
+    from the server and [None] when the server closes the channel or an error
+    occurs. Not thread-safe. *)
+
+  val unregister : 'a t -> callback_id -> unit
+  (** Unregister a callback previously registered with {!register}, which will
+      stop receiving new messages. No-op if the callback was unregistered before. *)
+
+  (**/**)
+
+  val wake : 'a t -> unit
+  (** Activate the handling loop, making sure the channel can receive messages.
+      No request will be sent. *)
+
+  val close : 'a t -> unit
+  (** [close c] closes the channel c. This function should be only use
```

Comment:
should only be used?

Reply:
I don't think this docstring is true anymore, there is no way to close the channel without also using `Eliom_bus`.
I made the `close` function public and fixed a potential memory leak in my last commit.